portal/app/models.py

```diff
@@ -145,3 +149,15 @@ class Meta:
 
     home = models.ForeignKey(Home)
     devices_json = JSONField(null=True, blank=True)
+
+
+class FarmDevice(models.Model):
+
+    class Meta:
+        db_table = 'farm_device_data'
+
+    home = models.ForeignKey(Home)
+    device_data = JSONField(null=True, blank=True)
+    device_uid = models.CharField(default = 'UNKNOWN', max_length=100)
```

Comment:
I think we should add a unique together on the device uid and prevent it from being blank or null.
Meaning we should force every farm device to have an ID that is unique within its location.
So something like:
```
class FarmDevice(models.Model):

    class Meta:
        db_table = 'farm_device_data'
        unique_together = ['home', 'device_uid']

    home = models.ForeignKey(Home)
    device_data = JSONField(null=True, blank=True)
    device_uid = models.CharField(max_length=100, blank=False, null=False)
....
```